jerry-core/ecma/operations/ecma-conversion.c

```diff
@@ -910,6 +912,119 @@ ecma_op_to_property_descriptor (ecma_value_t obj_value, /**< object value */
   return ret_value;
 } /* ecma_op_to_property_descriptor */
 
+/**
+ * ToInteger operation.
+ *
+ * See also:
+ *          ECMA-262 v5, 9.4
+ *          ECMA-262 v6, 7.1.4
+ *
+ * @return ECMA_VALUE_EMPTY if successful
+ *         conversion error otherwise
+ */
+ecma_value_t
+ecma_op_to_integer (ecma_value_t value, /**< ecma value*/
+                    ecma_number_t *number_p) /**< [out] ecma number */
+{
+  if (ECMA_IS_VALUE_ERROR (value))
+  {
+    return value;
+  }
+
+  /* 1 */
+  ecma_value_t to_number = ecma_get_number (value, number_p);
+
+  /* 2 */
+  if (ECMA_IS_VALUE_ERROR (to_number))
+  {
+    return to_number;
+  }
+
+  ecma_number_t number = *number_p;
+
+  /* 3 */
+  if (ecma_number_is_nan (number))
+  {
+    *number_p = ECMA_NUMBER_ZERO;
+    return ECMA_VALUE_EMPTY;
+  }
+
+  /* 4 */
+  if (ecma_number_is_zero (number) || ecma_number_is_infinity (number))
+  {
+    return ECMA_VALUE_EMPTY;
+  }
+
+  ecma_number_t floor_fabs = floor (fabs (number));
+
+  /* 5 */
+  *number_p = ecma_number_is_negative (number) ? -floor_fabs : floor_fabs;
+  return ECMA_VALUE_EMPTY;
+} /* ecma_op_to_integer */
+
+/**
+ * ToLength operation.
+ *
+ * See also:
+ *          ECMA-262 v6, 7.1.15
+ *
+ * @return ECMA_VALUE_EMPTY if successful
+ *         conversion error otherwise
+ */
+ecma_value_t
+ecma_op_to_length (ecma_value_t value, /**< ecma value*/
+                   uint32_t *length) /**< [out] ecma number */
+{
+  /* 1 */
+  if (ECMA_IS_VALUE_ERROR (value))
+  {
+    return value;
+  }
+
+#if ENABLED (JERRY_ES2015)
+  /* 2 */
+  ecma_number_t num;
+  ecma_value_t length_num = ecma_op_to_integer (value, &num);
+
+  /* 3 */
+  if (ECMA_IS_VALUE_ERROR (length_num))
+  {
+    return length_num;
+  }
+
+  /* 4 */
+  if (num <= 0.0f)
+  {
+    *length = 0;
+    return ECMA_VALUE_EMPTY;
+  }
+
+  /* 5 */
+  if (ecma_number_is_infinity (num))
```

Comment:
If we change this condition to `num >= (ecma_number_t) UINT32_MAX`, then the `<= 0.0` check above would no longer be necessary, the cast will handle negative numbers correctly.

Reply:
The 4th and 5th returns with a different values. the `<= 0.0` check returns with 0, while the `ecma_number_is_infinity` returns with `UINT32_MAX`.
I'm pretty sure those two are needed.

Reply:
The `(uint32_t)` cast correctly clamps negative numbers and negative infinity to zero. It also correctly clamps numbers > UINT32_MAX to UINT32_MAX. The only problem is +Infinity, which results in 0. Since `ecma_number_is_infinity` is also true for -Infinity, we can check >= UINT32_MAX  and return UINT32_MAX, which would also handle +Infinity correctly. This would achieve the same result with one less condition.

Reply:
My bad, apparently casting constant values work differently, than casting during runtime. It is necessray to check <= 0.